vsphere/vsphere.go

```diff
@@ -729,7 +742,9 @@ func (ops *vsphereOps) getDatastoreToUseInStoragePod(
 	}
 
 	spec.DeviceChange = deviceChange
+	ops.dsLock.LockWithKey(ops.cfg.VMUUID, storagePod.Name())
 	recommendedDatastore, err := recommendDatastore(ctx, vmObj, storagePod, spec)
+	ops.dsLock.UnlockWithKey(storagePod.Name())
```

Comment:
didn't quite get the lock with owner logic. Is the unlock with key sufficient? I assume it is, just trying to understand the way it works

Reply:
Owner is for debugging purposes. Key defines a single lock. Since the key is pod name, if there are more than one pod at some point, there will be different locks for each pod.